src/kirby/components/calendar/calendar.webview.html

```diff
@@ -0,0 +1,200 @@
+                    });
+                    return tableHeaderRow;
+                }
+
+                function getTableBody(month) {
+                    var tableBody = document.createElement('tbody');                    
+                    month.forEach(function(week) {
+                        tableBody.appendChild(getWeekRow(week));
+                    });
+                    tableBody.onclick = selectCell;
+                    return tableBody;
+                }
+
+                function getWeekRow(week) {
+                    var tr = document.createElement('tr');
+
+                    week.forEach(function(cell) {
+                        var dayCell = document.createElement('div');
+                        dayCell.className = cell.cssClasses + ' day-' + cell.date;
+                        dayCell.appendChild(document.createTextNode(cell.date));
+                        dayCell.dataset.isSelectable = cell.isSelectable;
+                        dayCell.dataset.day = cell.date;
+                        
+                        var td = document.createElement('td');
+                        td.appendChild(dayCell);
+                        tr.appendChild(td);
+                    });
+                    return tr;
+                }
+
+                function gotoPreviousMonth() {
+                    emitChangeMonth(-1);
+                }
+
+                function gotoNextMonth() {
+                    emitChangeMonth(1);
+                }
+
+                function selectCell(event) {
+                    var cellToSelect = event.target;
+                    var isSelectable = cellToSelect.dataset.isSelectable === 'true';
+                    var selectedDay = parseInt(cellToSelect.dataset.day);
+                    if (isSelectable && selectedDay) {
+                        setSelectedCell(cellToSelect);
+                        emitSelectedDay(selectedDay);
+                    }
+                }          
+
+                function setSelectedCell(cellToSelect) {
+                    var selectedCell = document.querySelector('#calendar .day.selected');
+                    if (selectedCell === cellToSelect) {
+                        return;
+                    }
+                    if (selectedCell) {
+                        selectedCell.classList.remove('selected');
+                    }
+                    if (cellToSelect) {
+                        cellToSelect.classList.add('selected');
+                    }
+                }
+
+                function setSelectedDay(day) {
+                    var cellToSelect = document.querySelector('#calendar .day.current-month.day-' + day);
+                    setSelectedCell(cellToSelect);
+                }
+
+                function init() {
+                    addNativeMsgListener();
+                }
+                init();
+            })();
+        </script>
+    </head>
+
+    <body style="margin: 0; padding: 0;">
```

Comment:
Consider moving this style to the css-file.

Reply:
It can give flicker until the css is loaded and parsed - that's why its added here